examples/api/lib/material/menu_bar/menu_bar.0.dart

```diff
@@ -0,0 +1,160 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for MenuBar
+
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+void main() => runApp(const MenuBarApp());
+
+enum MenuSelection {
+  about('About'),
+  showMessage('Show Message'),
+  hideMessage('Hide Message'),
+  colorMenu('Color Menu'),
+  colorRed('Red Background'),
+  colorGreen('Green Background'),
+  colorBlue('Blue Background'),
+  quit('Quit');
+
+  const MenuSelection(this.label);
+  final String label;
+}
+
+class MenuBarApp extends StatelessWidget {
+  const MenuBarApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      title: 'MenuBar Sample',
+      home: Scaffold(body: MyMenuBar()),
+    );
+  }
+}
+
+class MyMenuBar extends StatefulWidget {
+  const MyMenuBar({super.key});
+
+  @override
+  State<MyMenuBar> createState() => _MyMenuBarState();
+}
+
+class _MyMenuBarState extends State<MyMenuBar> {
+  bool get showingMessage => _showMessage;
+  bool _showMessage = false;
+  set showingMessage(bool value) {
+    if (_showMessage != value) {
+      setState(() {
+        _showMessage = value;
+      });
+    }
+  }
+
+  Color get backgroundColor => _backgroundColor;
+  Color _backgroundColor = Colors.red;
+  set backgroundColor(Color value) {
+    if (_backgroundColor != value) {
+      setState(() {
+        _backgroundColor = value;
+      });
+    }
+  }
+
+  void _activate(MenuSelection selection) {
+    switch (selection) {
+      case MenuSelection.about:
+        showAboutDialog(
+          context: context,
+          applicationName: 'MenuBar Sample',
+          applicationVersion: '1.0.0',
+        );
+        break;
+      case MenuSelection.showMessage:
+        showingMessage = true;
+        break;
+      case MenuSelection.hideMessage:
+        showingMessage = false;
+        break;
+      case MenuSelection.quit:
+        exit(0);
+      case MenuSelection.colorMenu:
+        break;
+      case MenuSelection.colorRed:
+        backgroundColor = Colors.red;
+        break;
+      case MenuSelection.colorGreen:
+        backgroundColor = Colors.green;
+        break;
+      case MenuSelection.colorBlue:
+        backgroundColor = Colors.blue;
+        break;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: <Widget>[
+        MenuBar(
+          menus: <MenuBarItem>[
+            MenuBarMenu(
+              autofocus: true,
+              label: 'Menu App',
+              menus: <MenuBarItem>[
```

Comment:
I've found it a little odd that the value of "menus" is a list of MenuBar_Items_.  Maybe "items" would be better? For menu bars that aren't native, it seems like apps might want to include other widgets (buttons, dividers, labels) rather than just menus.